packages/dcos-integration-test/extra/test_3dt.py

```diff
@@ -4,47 +4,58 @@
 import logging
 import os
 import tempfile
+import urllib.parse
 import zipfile
 
+import pytest
 import requests
 import retrying
 
 BASE_ENDPOINT_3DT = '/system/health/v1'
-PORT_3DT = 1050
-PORT_3DT_AGENT = 61001
 # Expected latency for all 3dt units to refresh after postflight
 LATENCY = 60
 
 
-def make_3dt_request(ip, endpoint, cluster, port=80):
+@pytest.fixture
+def make_3dt_request(cluster):
     """
-    a helper function to get info from 3dt endpoint. Default port is 80 for pulled data from agents.
-    if a destination port in 80, that means all requests should go though master (adminrouter) and we can re-use
-    cluster.get otherwise we can query 3dt agents directly to port 61001 (agent-adminrouter).
+    returns a function that performs authenticated connections to 3DT on the appropriate port
+    and appends ?cache=0 to all endpoints
     """
-    if port == 80:
+    def make_request(ip, endpoint, agent):
         assert endpoint.startswith('/'), 'endpoint {} must start with /'.format(endpoint)
-        logging.info('GET {}'.format(endpoint))
-        json_response = cluster.get(path=endpoint).json()
-        logging.info('Response: {}'.format(json_response))
+        endpoint = BASE_ENDPOINT_3DT + endpoint + '?cache=0'
+        scheme = urllib.parse.urlparse(cluster.dcos_uri).scheme
+        agent_port_map = {
+            'http': 61001,
+            'https': 61002}
+        master_port_map = {
+            'http': 80,
+            'https': 61002}
+        port = master_port_map.get(scheme)
+        if agent:
+            port = agent_port_map.get(scheme)
+        assert port, 'URL contains unrecognized scheme: {}'.format(cluster.dcos_uri)
+        url = '{}://{}:{}/{}'.format(scheme, ip, port, endpoint.lstrip('/'))
+        logging.info('GET {}'.format(url))
+        request_headers = {}
+        if cluster.auth_enabled:
+            request_headers = cluster.superuser_auth_header
+        request = requests.get(url, headers=request_headers)
+        assert request.ok
+        try:
+            json_response = request.json()
+            logging.info('Response: {}'.format(json_response))
+        except ValueError:
+            logging.error('Coult not deserialized json response from {}'.format(url))
```

Comment:
If you don't particularly mind, I'd like to avoid setting this via env var for now, as there are a few other cases like this and I would like to address them all uniformly